scala/README.md

```diff
@@ -68,6 +67,8 @@ Please read the [Scala Style Guide] carefully. The main points to consider are:
 
 0. Public methods must always have explicit return types.
 
+    While it is not required to annotate non-public members, remember that explicitly declaring the return type allows the compiler to verify correctness.
```

Comment:
what do we think about enforcing that all method definitions have type annotations? i definitely prefer that. it's annoying to look at someone's code later and try to figure out the return type of some helper function they wrote.